File: rust/examples/basic_implementation/example_delete.rs
#[path = "schema.rs"]
//Load the schema file
mod schema;

use skypydb::{json_map, ReactiveClient, Result};

fn main() -> Result<()> {
    //Create a new client
    let client = ReactiveClient::new("./db/_generated/skypydb_rust.db", None, None, None)?;
    let schema = schema::build_schema()?;

    //Get or create tables from the schema
    let tables = client.get_or_create_tables(&schema)?;

    //Error handling for tables
    let success_table = tables
        .get("success")
        .ok_or_else(|| skypydb::SkypydbError::table_not_found("Table 'success' not found"))?;

    //Delete data from the table
    let deleted = success_table.delete(json_map! {
        "component" => "AuthService",
        "user_id" => "user123",
    })?;

    //Print the results
    if deleted.is_empty() {
        println!("No deleted data found.");
    } else {
        for item in deleted {
            println!("{item}");
        }
    }

    Ok(())
}

File: demo/examples/python/basic_implementation/example_delete.py
import skypydb

# Create a client
client = skypydb.ReactiveClient()

# Create tables from the schema
# This reads the schema from db/schema.py and creates all tables
tables = client.get_or_create_table()
# if the tables already exists the programe get them instead

# Access your tables
success_table = tables["success"]
warning_table = tables["warning"]
error_table = tables["error"]

# delete data on the table assuming they are already data in the table.
success_table.delete(
    component="AuthService",
    user_id="user123"
)
